torch/_inductor/codegen/triton.py

```diff
@@ -1626,6 +1638,61 @@ def reduction_hint(node):
         else:
             return node.node.data.reduction_hint
 
+    @staticmethod
+    def can_use_32bit_indexing(numel: sympy.Expr, buffers: Iterable[ir.Buffer]) -> bool:
+        int_max = torch.iinfo(torch.int32).max
+        size_hint = V.graph.sizevars.size_hint
+        if size_hint(numel) > int_max:
+            return False
+
+        buf_sizes = [buf.get_layout().storage_size() for buf in buffers]
+        if any(size_hint(size) > int_max for size in buf_sizes):
+            return False
+
+        # Only install guards for 32-bit indexing as there is no correctness
+        # issue with using 64-bit for everything
+        for size in buf_sizes:
+            V.graph.sizevars.guard_leq(size, int_max)
```

Comment:
Wouldn't this break if you first pass to `(x.unsqueeze(0)+y).sum()` two vectors of size `2**15` and then two vectors of size `2**16`, as it wouldn't recompile?

Reply:
Agreed, I forgot to guard on numel